### PR TITLE
Fernschreiber learns filtering and searching

### LINKED ISSUE #9: add search function
currently (V.0.1) there's no search option like in the original Telegram client. Please add!

### PR DESCRIPTION
**Pull request by Wunderfitz**
These changes will fix #9. With this we will get:
- Filter capabilities for the chat list (use the magnifying glass next to the 'Fernschreiber' title
- Search messages in a chat (use the new pulley menu entry on chat page)
- Search public chats (use the new pulley menu entry on chat list page)

While implementing the features, I came across a few bugs which I fixed with this PR as well (I wanted to implement them directly on master, but I mixed up the branches...):
- Fernschreiber crashed when message deletion updates were received (there was an error in handling deletion ranges)
- Audio files didn't show album cover thumbnails properly (this is a leftover from the TDLib 1.7 migration)

### CODE REVIEW

File: qml/pages/SearchChatsPage.qml [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                                }
+                                onUserFullInfoReceived: {
+                                    if (foundChatListDelegate.isPrivateChat && userFullInfo["@extra"].toString() === foundChatListDelegate.foundChatInformation.type.user_id.toString()) {
+                                        foundChatListItem.tertiaryText.text = Emoji.emojify(userFullInfo.bio, foundChatListItem.tertiaryText.font.pixelSize, "../js/emoji/");
+                                    }
+                                }
+
+                                onBasicGroupFullInfoUpdated: {
+                                    if (foundChatListDelegate.isBasicGroup && groupId.toString() === foundChatListDelegate.foundChatInformation.type.basic_group_id.toString()) {
+                                        foundChatListItem.secondaryText.text = qsTr("%1 members").arg(Number(groupFullInfo.members.length).toLocaleString(Qt.locale(), "f", 0));
```
Comment (jgibbon) [inline qml/pages/SearchChatsPage.qml:178]: plural forms for members/subscribers missing

File: qml/pages/SearchChatsPage.qml [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                                        foundChatListItem.tertiaryText.text = Emoji.emojify(groupFullInfo.description, foundChatListItem.tertiaryText.font.pixelSize, "../js/emoji/");
+                                    }
+                                }
+                                onBasicGroupFullInfoReceived: {
+                                    if (foundChatListDelegate.isBasicGroup && groupId.toString() === foundChatListDelegate.foundChatInformation.type.basic_group_id.toString()) {
+                                        foundChatListItem.secondaryText.text = qsTr("%1 members").arg(Number(groupFullInfo.members.length).toLocaleString(Qt.locale(), "f", 0));
```
Comment (jgibbon) [inline qml/pages/SearchChatsPage.qml:184]: plural forms for members/subscribers missing

File: qml/pages/SearchChatsPage.qml [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                                        foundChatListItem.tertiaryText.text = Emoji.emojify(groupFullInfo.description, foundChatListItem.tertiaryText.font.pixelSize, "../js/emoji/");
+                                    }
+                                }
+
+                                onSupergroupFullInfoUpdated: {
+                                    if (foundChatListDelegate.isSupergroup && groupId.toString() === foundChatListDelegate.foundChatInformation.type.supergroup_id.toString()) {
+                                        if (foundChatListDelegate.relatedInformation.is_channel) {
+                                            foundChatListItem.secondaryText.text = qsTr("%1 subscribers").arg(Number(groupFullInfo.member_count).toLocaleString(Qt.locale(), "f", 0));
+                                        } else {
+                                            foundChatListItem.secondaryText.text = qsTr("%1 members").arg(Number(groupFullInfo.member_count).toLocaleString(Qt.locale(), "f", 0));
```
Comment (jgibbon) [inline qml/pages/SearchChatsPage.qml:194]: plural forms for members/subscribers missing

File: qml/pages/SearchChatsPage.qml [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                                        }
+                                        foundChatListItem.tertiaryText.text = Emoji.emojify(groupFullInfo.description, foundChatListItem.tertiaryText.font.pixelSize, "../js/emoji/");
+                                    }
+                                }
+                                onSupergroupFullInfoReceived: {
+                                    if (foundChatListDelegate.isSupergroup && groupId.toString() === foundChatListDelegate.foundChatInformation.type.supergroup_id.toString()) {
+                                        if (foundChatListDelegate.relatedInformation.is_channel) {
+                                            foundChatListItem.secondaryText.text = qsTr("%1 subscribers").arg(Number(groupFullInfo.member_count).toLocaleString(Qt.locale(), "f", 0));
+                                        } else {
+                                            foundChatListItem.secondaryText.text = qsTr("%1 members").arg(Number(groupFullInfo.member_count).toLocaleString(Qt.locale(), "f", 0));
```
Comment (jgibbon) [inline qml/pages/SearchChatsPage.qml:204]: plural forms for members/subscribers missing

File: qml/pages/SearchChatsPage.qml [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+                                    if (foundChatListDelegate.isBasicGroup && groupId.toString() === foundChatListDelegate.foundChatInformation.type.basic_group_id.toString()) {
+                                        foundChatListItem.secondaryText.text = qsTr("%1 members").arg(Number(groupFullInfo.members.length).toLocaleString(Qt.locale(), "f", 0));
+                                        foundChatListItem.tertiaryText.text = Emoji.emojify(groupFullInfo.description, foundChatListItem.tertiaryText.font.pixelSize, "../js/emoji/");
+                                    }
+                                }
+
+                                onSupergroupFullInfoUpdated: {
+                                    if (foundChatListDelegate.isSupergroup && groupId.toString() === foundChatListDelegate.foundChatInformation.type.supergroup_id.toString()) {
+                                        if (foundChatListDelegate.relatedInformation.is_channel) {
+                                            foundChatListItem.secondaryText.text = qsTr("%1 subscribers").arg(Number(groupFullInfo.member_count).toLocaleString(Qt.locale(), "f", 0));
```
Comment (jgibbon) [inline qml/pages/SearchChatsPage.qml:192]: plural forms for members/subscribers missing

File: qml/pages/SearchChatsPage.qml [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+                                        } else {
+                                            foundChatListItem.secondaryText.text = qsTr("%1 members").arg(Number(groupFullInfo.member_count).toLocaleString(Qt.locale(), "f", 0));
+                                        }
+                                        foundChatListItem.tertiaryText.text = Emoji.emojify(groupFullInfo.description, foundChatListItem.tertiaryText.font.pixelSize, "../js/emoji/");
+                                    }
+                                }
+                                onSupergroupFullInfoReceived: {
+                                    if (foundChatListDelegate.isSupergroup && groupId.toString() === foundChatListDelegate.foundChatInformation.type.supergroup_id.toString()) {
+                                        if (foundChatListDelegate.relatedInformation.is_channel) {
+                                            foundChatListItem.secondaryText.text = qsTr("%1 subscribers").arg(Number(groupFullInfo.member_count).toLocaleString(Qt.locale(), "f", 0));
```
Comment (jgibbon) [inline qml/pages/SearchChatsPage.qml:202]: plural forms for members/subscribers missing

**Review by jgibbon — COMMENTED** [review]
Stumbled upon a minor oversight while obviously not sleeping like I should. 

Happy new year!

**Comment by Wunderfitz** [comment]
> Stumbled upon a minor oversight while obviously not sleeping like I should.
> 
> Happy new year!

You're so right. Did the necessary changes with 2c078307dc1f61a145dedcdea4a5aeb842acc176. Happy new year to you as well! :)

**Pull request merged**